src/openindiana-about.py

```diff
@@ -39,216 +38,234 @@ def N_(message): return message
 VERSION   = "Development Version"
 LOCALEDIR = "%%DATADIR%%/locale"
 PIXMAPSDIR = "%%DATADIR%%/pixmaps"
-release_string = "OpenIndiana"
 
 copyright_string = N_("Copyright 2010 The OpenIndiana Project.\nCopyright 2010 Oracle Corporation and/or its affiliates.\nAll Rights Reserved. Use is subject to license terms.")
 
 release_text = N_("Release")
+system_text = N_("Hardware Platform")
 space_text = N_("Used Space")
 available_text = N_("Available Space")
 memory_text = N_("Memory")
 
+def get_version():
+        # This assumes standard OI /etc/release format
+        try:
+               with open('/etc/release', 'r') as f:
+                       line = f.readline().strip()
+                       return line.split(" ")[2]
+        except:
+               return VERSION
+
 def get_machine_info():
-	# This is gross, assumes the file output is regular
-	file_buffer = os.popen("/usr/sbin/prtdiag", "r").readlines()
-	if file_buffer is None:
-		return _("Unknown")
-	else:
-		machine_line = file_buffer.pop(0)
-		machine_name = machine_line.split("System Configuration: ",1)
-		return machine_name[1][:-1]
+        # This is gross, assumes the file output is regular
+        file_buffer = os.popen("/usr/sbin/prtdiag", "r").readlines()
+        if file_buffer is None:
+                return _("Unknown")
+        else:
+                machine_line = file_buffer.pop(0)
+                machine_name = machine_line.split("System Configuration: ",1)
+                return machine_name[1][:-1]
 
 def get_machine_memory():
-	# This is also gross, assumes the file output is regular
-	file_buffer = os.popen("/usr/sbin/prtconf", "r").readlines()
-	memory_line = file_buffer.pop(1)
-	memory_info = memory_line.split("Memory size: ", 1)
-	labels = memory_info[1][:-1].split()
-	value = labels[0];
-	unit = labels[1];
-	if (re.compile("Megabytes").match(unit)):
-		return _("%.1f MB") % string.atoi(value)
-	elif (re.compile("Gigabytes").match(unit)):
-		return _("%.1f GB") % string.atoi(value)
-	else:
-		return value + " " + unit;
+        # This is also gross, assumes the file output is regular
+        file_buffer = os.popen("/usr/sbin/prtconf", "r").readlines()
+        memory_line = file_buffer.pop(1)
+        memory_info = memory_line.split("Memory size: ", 1)
+        labels = memory_info[1][:-1].split()
+        value = labels[0];
+        unit = labels[1];
+        if (re.compile("Megabytes").match(unit)):
+                return _("%.1f MB") % int(value)
+        elif (re.compile("Gigabytes").match(unit)):
+                return _("%.1f GB") % string.atoi(value)
```

Comment:
Why `string.atoi()` remains in one line, but was changed to `int()` in another? Miss or intent?